stl/inc/yvals_core.h

```diff
@@ -799,7 +799,21 @@
 #define _DEPRECATE_STDEXT_HASH_UPPER_BOUND
 #endif // ^^^ warning disabled ^^^
 
-// next warning number: STL4024
+// P0966R1 [depr.string.capacity]
+#if _HAS_CXX20 && !defined(_SILENCE_CXX20_STRING_RESERVE_DEFAULT_ARGUMENT_DEPRECATION_WARNING) \

```

Comment:
As mentioned in my reply to Casey's typo comment, C++20 doesn't depict a default argument, so naming this `STRING_RESERVE_DEFAULT_ARGUMENT` seems confusing. `STRING_NULLARY_RESERVE` would be more accurate, although perhaps "nullary" isn't a very common term. `STRING_RESERVE_WITHOUT_ARGUMENTS` would be better, I think.

Reply:
(The deprecation macro and silencing escape hatch should change together, of course.)